devmand/gateway/src/devmand/channels/cli/SshSocketReader.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2019-present, Facebook, Inc.
+// All rights reserved.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree. An additional grant
+// of patent rights can be found in the PATENTS file in the same directory.
+
+#include <devmand/channels/cli/SshSocketReader.h>
+#include <devmand/channels/cli/SshSessionAsync.h>
+#include <event2/event.h>
+#include <thread>
+#include <chrono>
+
+using devmand::channels::cli::sshsession::SshSessionAsync;
+
+void sshReadNotificationThread(struct event_base *base);
+void sshReadNotificationThread(struct event_base *base) {
+    while (true) {
+        int rv = event_base_dispatch(base);
+        if (rv == 1) {
+            std::this_thread::sleep_for(std::chrono::seconds(1));
+            continue;
+        }
+        else {
+            const char *error = "event_base_dispatch catastrophic failure, terminating";
+            MLOG(MERROR) << error << " returned value: " << rv;
+            throw std::runtime_error(error);
+        }
+    }
+}
+
+devmand::channels::cli::SshSocketReader::SshSocketReader() {
+    base = event_base_new();
```

Comment:
not now but at some point maybe we can tie this into a folly Event base?

Reply:
Yes, I have to check the docs for folly Event base to see what it supports